src/qa4sm_reader/handlers.py

```diff
@@ -55,12 +55,18 @@ def get_varmeta(self) -> (tuple, tuple, tuple):
             this is the dataset for which the metric is calculated
         dss : id, dict
             this is the additional dataset in TC variables
+        scale_ds: id, dict
+            this is the scaling dataset
         """
         if self.g == 0:
             ref_ds = self.Datasets.dataset_metadata(self.Datasets._ref_id())
-            mds, dss = None, None
+            mds, dss, scale_ds = None, None, None
 
         else:
+            scale_ds = None
+            if 'val_scaling_ref' in self.Datasets.meta.keys():
+                scale_ref_id = int(self.Datasets.meta['val_scaling_ref'][-1:])
```

Comment:
is there a way without hard coding the attribute key?

Reply:
also, I know at the moment there are not more than 6 datasets, but still I wouldnt get the ID this way. It only works for single digit numbers

Reply:
I'll check if i find something in the objects, to my knowledge the only  indication from the netcdf is via this attribute key, but maybe its already checked somewhere else. 

sure I'll change the way to fetch the ID.